src/tally.rs

```diff
@@ -163,21 +180,36 @@ async fn run_days(
     number_of_runs: usize,
 ) -> Result<Vec<(u32, (usize, Option<usize>))>, AocError>
 {
-    let mut set = JoinSet::new();
-    let mut vec: Vec<(u32, (usize, Option<usize>))> = Vec::new();
     let cargo_folder = cargo_path().await?;
-    for day in days
-    {
-        let cargo_folder = cargo_folder.clone();
-        set.spawn(async move { (day.1, run_day(cargo_folder, day, number_of_runs).await) });
-    }
+    let multi = MultiProgress::new();
+    let sty = ProgressStyle::with_template(
+        "[{elapsed_precise}] {msg}... {bar:40.cyan/blue} {pos:>7}/{len:7}",
+    )
+    .unwrap()
+    .progress_chars("##-");
+
+    // Sort it to get the progress bars in increasing order
+    let mut days = days;
+    days.sort_unstable_by_key(|k| k.1);
+    let days = days
+        .into_iter()
+        .map(|day| {
+            let sty = sty.clone();
+            let progress = multi.add(ProgressBar::new(number_of_runs as u64));
+
+            progress.set_style(sty);
+            progress.set_message(format!("Running day {}", day.1));
+            (day, progress)
+        })
+        .collect::<Vec<_>>();
 
-    while let Some(Ok((day, res))) = set.join_next().await
-    {
-        vec.push((day, res?));
-    }
+    Ok(thread_exec(days, |(day, progress)| {
+        let cargo_folder = cargo_folder.clone();
+        let d = day.1;
 
-    Ok(vec)
+        let res = run_day(cargo_folder, &day, number_of_runs, progress).expect("Running day");
+        (d, res)
+    }))
 }
 
 fn print_info(days: Vec<(u32, (usize, Option<usize>))>, not_done: Vec<u32>, number_of_runs: usize)
```

Comment:
Not in the diff here, but I don't think we need to sort the vec created in `print_info`(?)

Reply:
(I renamed _vec to data)
If you mean this line
```rs
let mut data: Vec<_> = vec.iter().map(|(_, time)| *time).collect();
data.sort_unstable();
```
Then I sort it to get the median
```rs
let median = data[data.len() / 2];
```

Reply:
> Then I sort it to get the median

Ah yes of course, my bad 😅